db/src/tx.rs

```diff
@@ -254,13 +254,13 @@ impl<'conn, 'a> Tx<'conn, 'a> {
 
                     let v = match v {
                         entmod::AtomOrLookupRefOrVectorOrMapNotation::Atom(v) => {
-                            if attribute.value_type == ValueType::Ref && v.is_text() {
-                                Either::Right(LookupRefOrTempId::TempId(temp_ids.intern(v.as_text().cloned().map(TempId::External).unwrap())))
+                            if attribute.value_type == ValueType::Ref && v.inner.is_text() {
```

Comment:
Can we simplify some of this by making `ValueAndSpan` implement `Deref<Value>`? That way you can treat it like a `Value` for method call purposes…